examples/with-nextjs-and-clerk-auth/src/lib/clerk-api/set-current-user-private-metadata.ts

```diff
@@ -0,0 +1,20 @@
+import { clerkClient, currentUser } from '@clerk/nextjs';
+
+import { PrivateMetadata } from '@/lib/clerk-api/types';
+
+export const setCurrentUserPrivateMetadata = async (
+  privateMetadata: Partial<PrivateMetadata>
+): Promise<void> => {
+  const user = await currentUser();
+
+  if (user === null) {
+    throw new Error('No current user available');
+  }
+
+  await clerkClient.users.updateUserMetadata(user.id, {
+    privateMetadata: {
+      ...user?.privateMetadata,
+      ...privateMetadata,
```

Comment:
One non-obvious thing about Meta Data: you don't need to pass the previous value. 

If you need to just create/update a single property you can simply call:
```tsx
await clerkClient.users.updateUserMetadata(user.id, {
    privateMetadata: {new_field: 'my-data'}
})
```

In this case, other properties will not be overwritten:
```json
{ "new_field": "my-data", "previous_field": "prev-data"}
```

Once you need to remove the field, just call:

```tsx
await clerkClient.users.updateUserMetadata(user.id, {
    privateMetadata: {new_field: null}
})
```

After that, Metadata will look like: `{"previous_field": "prev-data"}`